charts/gateway-helm/templates/certgen-rbac.yaml

```diff
@@ -62,3 +62,63 @@ subjects:
 - kind: ServiceAccount
   name: '{{ include "eg.fullname" . }}-certgen'
   namespace: '{{ .Release.Namespace }}'
+---
+{{- if .Values.topologyInjector.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: '{{ include "eg.fullname" . }}-certgen:{{ .Release.Namespace }}'
+  labels:
+  {{- include "eg.labels" . | nindent 4 }}
+  {{- if .Values.certgen.rbac.labels }}
+  {{- toYaml .Values.certgen.rbac.labels | nindent 4 }}
+  {{- end }}
+  annotations:
+    "helm.sh/hook": pre-install, pre-upgrade
+    "helm.sh/hook-weight": "-1"
```

Comment:
can we rm the weight annotation ? dont see it on any certgen resources

Reply:
It's necessary here due to helm resource sorting. Otherwise the cert gen job executes before the ClusterRole is created.

Reply:
thinking out loud - why dont the other resources in this file cause the same issue

Reply:
It's the helm resource sorting and in the test log you can see the clusterrol/rolebinding never get installed - https://github.com/envoyproxy/gateway/actions/runs/14654407193/job/41127299304?pr=5352